File: go/fizzbuzz.go
package main

import "fmt"
import "strconv"

func fizzbuzzer(num int, replacements map[int]string) string {
	out := ""

	for multiple, replacement := range replacements {
		if num%multiple == 0 {
			out += replacement
		}
	}

	// if `out` is empty, there are no replacements; return stringified `num`
	if len(out) == 0 {
		return strconv.Itoa(num)
	}

	return out
}

func main() {
	replacement_map := map[int]string{
		3: "Fizz",
		5: "Buzz",
	}

	for num := 1; num <= 100; num++ {
		fmt.Println(fizzbuzzer(num, replacement_map))
	}
}

File: rust/fizzbuzz.rs
struct ReplacementMapEntry {
    multiple: i32,
    replacement: &'static str
}

fn fizzbuzzer(num: i32, replacements: &[ReplacementMapEntry]) -> String {
    let mut out = String::new();

    for i in 0..replacements.len() {
        if num % replacements[i].multiple == 0 {
            out.push_str(replacements[i].replacement);
        }
    }

    if out.is_empty() {
        return format!("{}", num);
    }

    return out;
}

pub fn main() {
    let replacements: [ReplacementMapEntry; 2] = [
        ReplacementMapEntry {multiple: 3, replacement: "Fizz"}, 
        ReplacementMapEntry {multiple: 5, replacement: "Buzz"}
    ];
        
    for num in 1..=100 {
        println!("{}", fizzbuzzer(num, &replacements));
    }
}
